src/Simplex/Messaging/Agent/Store/SQLite/Schema.hs

```diff
@@ -77,27 +77,39 @@ connections =
       snd_host TEXT,
       snd_port TEXT,
       snd_id BLOB,
-      PRIMARY KEY(conn_alias),
-      FOREIGN KEY(rcv_host, rcv_port, rcv_id) REFERENCES rcv_queues(host, port, rcv_id),
-      FOREIGN KEY(snd_host, snd_port, snd_id) REFERENCES snd_queues(host, port, snd_id)
+      last_rcv_msg_id INTEGER,
+      last_snd_msg_id INTEGER,
+      PRIMARY KEY (conn_alias),
+      FOREIGN KEY (rcv_host, rcv_port, rcv_id) REFERENCES rcv_queues (host, port, rcv_id),
+      FOREIGN KEY (snd_host, snd_port, snd_id) REFERENCES snd_queues (host, port, snd_id)
     ) WITHOUT ROWID;
   |]
 
 messages :: Query
 messages =
   [sql|
     CREATE TABLE IF NOT EXISTS messages(
-      agent_msg_id INTEGER NOT NULL,
+      agent_msg_id INTEGER PRIMARY KEY,
+      agent_timestamp TEXT NOT NULL,
       conn_alias TEXT NOT NULL,
-      timestamp TEXT NOT NULL,
+      rcv_msg_id INTEGER,
+      snd_msg_id INTEGER,
```

Comment:
why do we need two IDs? would one ID and direction be better? I mean it made sense if we had two tables and one to unify them, but now we have one?

Reply:
not sure :)

Reply:
yeah.. makes sense, that would be less messy. I don't know why I didn't revert them